tests/php/core-api/wpcom-fields/test-post-fields-publicize-connections.php

```diff
@@ -0,0 +1,294 @@
+    // WP_Test_Jetpack_REST_Testcase::setUp()'s `do_action( 'rest_api_init' )`.
+    // Thus, the order would normally be:
+    // 1. wpcom_rest_api_v2_load_plugin( 'WPCOM_REST_API_V2_Post_Publicize_Connections_Field' ) (calls `add_action( 'rest_api_init' )`)
+    // 2. WP_Test_Jetpack_REST_Testcase::setUp() (calls `do_action( 'rest_api_init' )`)
+    // 3. WP_UnitTestCase::setUp() (calls `WP_UnitTestCase::_backup_hooks()`).
+    // and we end up incorrectly backing up the hooks we've changeed via
+    // WPCOM_REST_API_V2_Post_Publicize_Connections_Field's `add_action( 'rest_api_init' )` call.
+    // (We want to backup the "default" hooks prior to any changes these tests make.)
+    //
+    // Instead, we "manually" call WP_UnitTestCase::_backup_hooks() first:
+    // 1. WP_UnitTestCase::_backup_hooks()
+    // 2. wpcom_rest_api_v2_load_plugin( 'WPCOM_REST_API_V2_Post_Publicize_Connections_Field' ) (calls `add_action( 'rest_api_init' )`)
+    // 3. WP_Test_Jetpack_REST_Testcase::setUp() (calls `do_action( 'rest_api_init' )`)
+    // 4. WP_UnitTestCase::setUp() (is smart enough *not* to call `WP_UnitTestCase::_backup_hooks()` a second time).
+    // and we are now correctly backing up the default hooks so that when we restore hooks
+    // in WP_UnitTestCase::tearDown(), we correctly restore the hooks as they were prior to
+    // WPCOM_REST_API_V2_Post_Publicize_Connections_Field's `add_action( 'rest_api_init' )` call.
+    $this->_backup_hooks();
+
+    // The Publicize Connections field is loaded conditionally based on whether
+    // the Publicize Module is active.
+    // The Module is not active in this test environment, so load it manually.
+    // It might look like it would be simpler if we only did this once in
+    // ::wpSetUpBeforeClass() instead of once for eeach test here in ::setUp().
+    // If we did that, though, we'd lose (and break) WP_UnitTestCase's
+    // hook backup/restore functionality (see comment on above line).
+    wpcom_rest_api_v2_load_plugin( 'WPCOM_REST_API_V2_Post_Publicize_Connections_Field' );
+  }
+
+  public function tearDown() {
+		global $wpcom_rest_api_v2_plugins;
+
+    parent::tearDown();
+
+    wp_delete_post( $this->draft_id, true );
+
+    if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+      // WPCOM_REST_API_V2_Post_Publicize_Connections_Field is already active
+      // in this test environment. No need for the fancy hoops further below.
+      return;
+    }
+
+    // De-memoize wpcom_rest_api_v2_load_plugin()
+    unset( $wpcom_rest_api_v2_plugins['WPCOM_REST_API_V2_Post_Publicize_Connections_Field'] );
+
+    // Restore this core global
+    if ( 'unset' === $this->_backup_wp_rest_additional_fields ) {
+      unset( $GLOBALS['wp_rest_additional_fields'] );
+    } else {
+      $GLOBALS['wp_rest_additional_fields'] = $this->_backup_wp_rest_additional_fields;
+    }
+
+	// Clean up custom meta from publicizeable post types
+	foreach ( get_post_types() as $post_type ) {
```

Comment:
Should we apply this cleanup for the `-inactive` test class too?

Reply:
I would be happy to know the answer, but I don't know :) This specific test was adapted to work in WPCOM, so it's `setUp`/`tearDown` logic is a bit different from `-inactive` test file. Since `inactive` is not needed on WPCOM, we might not need to update it. 

@mdawaffe maybe you have some thoughts on this?